### PR TITLE
fixed compile issue on macOS 12

### PR DESCRIPTION
**Pull request by Code-Hex**
added build stage for macOS 11 and 12

## Which issue(s) this PR fixes:

Relate https://github.com/Code-Hex/vz/issues/170

### CODE REVIEW

**Comment by Code-Hex** [comment]
Oh, I need to check it

https://github.com/Code-Hex/vz/actions/runs/11675552277/job/32510321947?pr=171

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -28,6 +28,8 @@ jobs:
       fail-fast: false
       matrix:
         os:
+        - macos-11  # Intel
+        - macos-12  # Intel
```
Comment (AkihiroSuda) [inline .github/workflows/compile.yml:32]: macos-11 is already dead and 12 will be removed soon
- https://github.com/actions/runner-images/issues/10721

Comment (Code-Hex) [reply .github/workflows/compile.yml:32]: Thanks information! I didn't know that.
It's tough to lose the means of detecting the problems that occurred this time.

**Comment by Code-Hex** [comment]
I confirmed to be compilable on macOS 12
- [1.22](https://github.com/Code-Hex/vz/actions/runs/11676269835/job/32512250474)
- [1.23](https://github.com/Code-Hex/vz/actions/runs/11676269835/job/32512250653)

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**